test/built-ins/TypedArrays/of/new-instance-using-custom-ctor.js

```diff
@@ -0,0 +1,27 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+id: sec-%typedarray%.of
+description: >
+  Return a new TypedArray using a custom Constructor
+includes: [testTypedArray.js]
+---*/
+
+testWithTypedArrayConstructors(function(TA) {
+  var called = 0;
+
+  var ctor = function(len) {
+    called++;
```

Comment:
The counter should be incremented *after* the assertion. That way, if the assertion fails and the runtime swallows the completion, we'll know something went wrong.